✨ Implement In-Editor AI Feature: Generate group titles with AI

basuabhirup opened the pull request:
This PR is now ready to be reviewed. This attempts to resolve #1577 .

This is the most basic implementation, and there are scopes for further improvements like allowing the user to select specific models / set usage limits etc. 


https://github.com/user-attachments/assets/09e7d920-ea01-4611-ac17-970f444b246d

baptisteArno commented on packages/prisma/postgresql/schema.prisma:
```diff
@@ -122,6 +124,17 @@ model WorkspaceInvitation {
   workspace   Workspace     @relation(fields: [workspaceId], references: [id], onDelete: Cascade)
 }
 
+model WorkspaceAiFeature {
+  id           String       @id @default(cuid())
+  createdAt    DateTime     @default(now())
+  updatedAt    DateTime     @default(now()) @updatedAt
+  workspace    Workspace    @relation(fields: [workspaceId], references: [id], onDelete: Cascade)
+  workspaceId  String
+  prompt       String
+  credential   Credentials? @relation(fields: [credentialId], references: [id], onDelete: SetNull)
+  credentialId String?
+}
+
```
Is it really necessary? I would probably just inline that in the Workspace model. Typebot has not a relation-heavy db schema.

baptisteArno replied:
Yes, let's keep it simple and have one field for group title auto rename. Let's build for it without thinking of next AI features this should just implement #1577

In Typebot, defaults are always constants in the code.

basuabhirup replied:
Okay, got it 💯 Trying to keep it simple and focusing on implementing #1577 for now. I will share my progress and revert back here if any further help is needed from you. Thank you @baptisteArno

basuabhirup commented:
@baptisteArno The user flow to set / update credentials and prompts for the AI features is implemented with minimum changes in the workspace schema. Please provide your feedback on this implementation from the UX point of view.

https://github.com/user-attachments/assets/d15a812d-bbcd-4636-b65f-bcdea08ff7e2

I'm now working on the main part of the feature i.e. generating group titles, and will keep you updated here.

baptisteArno commented:
> @baptisteArno The user flow to set / update credentials and prompts for the AI features is implemented with minimum changes in the workspace schema. Please provide your feedback on this implementation from the UX point of view.

Awesome to see! 🙏 I suggest you focus on making the feature work, I will improve the UI / UX afterwards if it is ok with you? From what I can see for now, this is nice 👍

basuabhirup commented:
> Awesome to see! 🙏 I suggest you focus on making the feature work, I will improve the UI / UX afterwards if it is ok with you? From what I can see for now, this is nice 👍

Yeah, sure sure!

basuabhirup commented:
@baptisteArno The most basic version of this feature is implemented, although there are further scopes of improvements. Please note that:

1. I haven't tested the Anthropic API as I don't have any credits left for now and so I would request you to kindly check once if the current feature is working properly with an Anthropic API key or not.
2. Model names are currently hard-coded for each AI provider in the backend, but we can allow the users to select models as well.
3. The Workspace schema is updated with 3 new additional fields, so please migrate your db before testing.


https://github.com/user-attachments/assets/89e33d25-8ffd-4b06-ba70-21e0b6619cea

baptisteArno commented:
That looks awesome. Thanks for working on that. 🔥

baptisteArno commented:
Alright so I simplified a lot of things. You probably did not know but we have a bunch of useful components like `DropdownList`, `CredentialsDropdown` and `CredentialsCreateModal`.

Also removed the loading state which is not really useful since we use small models.

basuabhirup commented:
> Alright so I simplified a lot of things. You probably did not know but we have a bunch of useful components like `DropdownList`, `CredentialsDropdown` and `CredentialsCreateModal`.
> 
> Also removed the loading state which is not really useful since we use small models.

@baptisteArno Ahh I see. Got it. Thanks for the clarification!
